python/semantic_kernel/connectors/memory/chroma/chroma_memory_store.py

```diff
@@ -100,11 +100,14 @@ async def get_collection_async(
         self, collection_name: str
     ) -> Optional["Collection"]:
         try:
-            # Current version of ChromeDB rejects camel case collection names.
-            return self._client.get_collection(
-                name=camel_to_snake(collection_name),
-                embedding_function=self._default_embedding_function,
-            )
+            if camel_to_snake(collection_name) in await self.get_collections_async():
```

Comment:
I see we're adding the preemptive check to see if the collection exists by calling `await self.get_collections_async()` to handle the issue where the remote instance returns an `HTTPError` instead of a `ValueError`. Now that we're making that call every time we want to get a single collection, is this introducing performance overhead? Thinking about the case where the collection is long or if this is called frequently, it could become a bottleneck, right?